File: preferences-annotations/src/main/java/com/autopreferences/annotations/PreferenceKey.java
package com.autopreferences.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks field as SharedPreference Key that will be used in SharedPreference.putString("key", value)
 */
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.FIELD)
public @interface PreferenceKey {
	String value();

	boolean generateDefaultOverloadMethod() default true;
}

File: preferences-annotations/src/main/java/ru/starksoft/autopreferences/PreferenceKey.kt
package ru.starksoft.autopreferences

/**
 * Marks field as SharedPreference Key that will be used in SharedPreference.putString("key", value)
 */
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.FIELD)
annotation class PreferenceKey(val value: String, val generateDefaultOverloadMethod: Boolean = true)